### DISCUSSION TITLE
Seeking Advice: Optimizing Dynamic Style Implementation in MapLibre

### DISCUSSION THREAD

**Discussion by ArbyC** [maplibre/maplibre-gl-js#3602] [General]
I'm part of a development team that has recently transitioned from OpenLayers to MapLibre. Our primary motivation for this switch was to leverage MapLibre's advanced 3D features and its performance benefits. While we've been excited about these new capabilities, we've encountered some challenges, particularly around implementing dynamic styling for map layers.

In OpenLayers, we had a streamlined process for adding and updating styles dynamically, but the transition to MapLibre's styling specifications has proven to be more complex than anticipated. We are finding that managing dynamic styles in MapLibre, especially when dealing with data-driven styling and real-time updates, is not as straightforward

**Comment by HarelM** [comment]
While I generally agree that dynamic styling is complicated is maplibre, I'm not sure I fully understand the question. Can you elaborate?
In recent versions we updated the set style method and style diff capabilities which should help with dynamic styling.
Having said that, the openlayers "layers" (grouping) doesn't have a good alternative in maplibre
